ts/src/graphql/query/edge_connection.ts

```diff
@@ -30,7 +32,7 @@ export class GraphQLEdgeConnection<
   TEdge extends Data,
   TViewer extends Viewer = Viewer,
 > {
-  query: EdgeQuery<TSource, Ent, TEdge>;
+  query: MaybePromise<EdgeQuery<TSource, Ent, TEdge>>;
```

Comment:
might be simpler to take a query here and update everything to work on Promise<T>

so this.query = Promise.resolve(query) and everything below uses that same query?

what happens when you create multiple `Promise.query` on the same instance multiple times? we end up calling the function multiple times right?

I think you need a test for this case that handles when it's passed a Promise instead of just the function

there's also more code that's going to need to be updated since the value of query is different now

see https://github.com/lolopinto/ent/actions/runs/10712869234/job/29703982341?pr=1829

Reply:
> so this.query = Promise.resolve(query) and everything below uses that same query?

Oh that's a good idea.

> what happens when you create multiple `Promise.query` on the same instance multiple times? we end up calling the function multiple times right?

Assuming you mean `Promise.resolve`, If you call `Promise.resolve(Promise.resolve(somePromise))`, it's the equivalent of doing `await (await somePromise)`, the outer one is just a no-op since the promise is already resolved.

> I think you need a test for this case that handles when it's passed a Promise instead of just the function

👍  I'll get on that today

> there's also more code that's going to need to be updated since the value of query is different now
> see https://github.com/lolopinto/ent/actions/runs/10712869234/job/29703982341?pr=1829

Thanks. I'll do a more thorough check to make sure there aren't any others as well.

Reply:
> what happens when you create multiple Promise.query on the same instance multiple times? we end up calling the function multiple times right?

Referring to Promise.resolve(query) N times. Does it end up running the underlying await function multiple then? Do we need some kind of memoize? Or we just leave that out of scope here?

Reply:
It does not end up calling it multiple times, once a promise is resolved, calling resolve on it again just returns the resolved value immediately.

Reply:
But actually with your suggestion of making `this.query` always a promise, I'm no longer calling `Promise.resolve` multiple times anyways. 👍

Reply:
Ok, then in this case doing the Promise once at the beginning seems like a good thing since previous implementation would have created a new Promise.resolve at multiple call sites?